src/node_config.rs

```diff
@@ -191,6 +235,67 @@ pub fn cli_parse() -> CliConfig {
             .help("Disable LAN discovery"))
         .get_matches();
 
+    match matches.subcommand() {
+        ("config", Some(m)) => run_config(m),
+        _ => run_args(&matches),
+    }
+}
+
+/// Parse settings from a saved file.
+fn parse_config(config_path: String) ->Result<NodeConfig, ()> {
+    let mut settings = Config::default();
+
+    let default_str = r#"
+        log_type: Stderr
+        keys_file: keys
+        udp_addr: 0.0.0.0:33445
+        tcp_addrs:
+          - 0.0.0.0:33445
+        motd: "{{start_date}} {{uptime}} Tcp: incoming {{tcp_packets_in}}, outgoing {{tcp_packets_out}}, Udp: incoming {{udp_packets_in}}, outgoing {{udp_packets_out}}"
+        bootstrap_nodes:
+          - pk: 1D5A5F2F5D6233058BF0259B09622FB40B482E4FA0931EB8FD3AB8E7BF7DAF6F
+            addr: 198.98.51.198:33445
+          - pk: DA4E4ED4B697F2E9B000EEFE3A34B554ACD3F45F5C96EAEA2516DD7FF9AF7B43
+            addr: 185.25.116.107:33445
+        lan_discovery_enabled: True
+        sk_passed_as_arg: False
+        threads_config: auto
+        "#;
+
+    let default_file = CfgFile::from_str(default_str, FileFormat::Yaml);
+    settings.merge(default_file).map_err(|_| ())?;
+
+    let config_file = if !Path::new(&config_path).exists() {
+        panic!("Can't find config file {}", config_path);
+    } else {
+        CfgFile::with_name(&config_path)
+    };
+
+    settings.merge(config_file).map_err(|_| ())?;
```

Comment:
Why map this error to unit type? Doing that we are losing error message. Can we use common error type from all error cases? If no perhaps we should just panic in place?

Reply:
panic is OK

Reply:
done